lib/shopify_cli/environment.rb

```diff
@@ -106,7 +106,17 @@ def self.infer_spin?(env_variables: ENV)
 
     def self.spin_url(env_variables: ENV)
       if infer_spin?(env_variables: env_variables)
-        %x(spin info fqdn 2> /dev/null).strip
+        # TODO: Remove version check and delete spin-legacy branch
+        #  once spin2 becomes the installed "spin" binary by default
+        spin_version = %x(spin version 2> /dev/null).strip
+        if spin_version.start_with?("spin-")
+          # spin2
+          raise "SPIN_INSTANCE must be specified" unless ENV.key?("SPIN_INSTANCE")
```

Comment:
I am unsure if this should be a raised exception (which includes the stack trace being printed), or if there exists a better existing error-and-exit mechanism so that only a single line error message gets outputted.

Reply:
Use the following instead:

```rb
raise ShopifyCLI:: Abort, "SPIN_INSTANCE must be specified"
```

Those errors are automatically handled at the root of the CLI.